### PR TITLE
Show number of ready containers

### LINKED ISSUE #539: [UI] Clear indication of pods with containers not ready
The UI does not show pod's readiness well. It should. 

NOTE that readiness a count of all containers in the pod, not just a true/false value.

```
argo-artifacts-minio-79f65cfcb6-x2xk5    1/1       Running            0          27d
guestbook-ui-5bb569f586-xq9zk            1/1       Running            0          30d
```

### PR DESCRIPTION
**Pull request by merenbach**
Shows count of ready containers and closes https://github.com/argoproj/argo-cd/issues/539.

image

image

### CODE REVIEW

File: src/app/applications/components/application-resources-tree/application-resources-tree.tsx [hunk 1]
```diff
@@ -29,6 +29,17 @@ function getGraphSize(nodes: dagre.Node[]): { width: number, height: number} {
     return {width, height};
 }
 
+// CountReadyContainerStatuses takes a list of container statuses and counts the running ones.
+function countReadyContainerStatuses(containerStatuses) {
```
Comment (alexmt) [inline src/app/applications/components/application-resources-tree/application-resources-tree.tsx]: Please fix build error:

```
yarn build


ERROR in [at-loader] ./src/app/applications/components/application-resources-tree/application-resources-tree.tsx:33:38
    TS7006: Parameter 'containerStatuses' implicitly has an 'any' type.

ERROR in [at-loader] ./src/app/applications/components/application-resources-tree/application-resources-tree.tsx:35:32
    TS7006: Parameter 'containerStatus' implicitly has an 'any' type.
Child html-webpack-plugin for "index.html":
```

File: src/app/applications/components/application-resources-tree/application-resources-tree.tsx [hunk 2]
```diff
@@ -142,6 +153,9 @@ export const ApplicationResourcesTree = (props: {
                         <div className='application-resources-tree__node-labels'>
                             {props.nodeLabels && props.nodeLabels(node).map((label) => <span key={label}>{label}</span>)}
                             <span>{kubeState.kind}</span>
+                            {kubeState.status && kubeState.status.containerStatuses && (
```
Comment (alexmt) [inline src/app/applications/components/application-resources-tree/application-resources-tree.tsx]: We already have code which creates labels for pods. It is good to keep code with similar responsibility in one please. Please move your changes to https://github.com/argoproj/argo-cd-ui/blob/f9c39fbc3b2e8c64fe19dd74593d8dcb1e439cc9/src/app/applications/components/application-details/application-details.tsx#L352

**Review by alexmt — CHANGES_REQUESTED** [review]

**Comment by merenbach** [comment]
Great calls, @alexmt!  I believe they are now addressed!

File: src/app/applications/components/application-details/application-details.tsx [hunk 3]
```diff
@@ -357,6 +362,11 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{ na
             if (reason) {
                 labels.push(reason);
             }
+
+            const allStatuses = resourceNode.state.status.containerStatuses;
+            const readyStatuses = allStatuses.filter((s: ContainerStatus) => ('running' in s.state));
+            const readyContainers = "" + readyStatuses.length + "/" + allStatuses.length + " containers ready";
```
Comment (alexmt) [inline src/app/applications/components/application-details/application-details.tsx]: Please fix linter warnings:

```
yarn lint                                                                                                                        (tools-sgmnt-prod.cluster.k8s.local/dev-deploy-argocd-us-west-2-prod-cdp)
yarn run v1.6.0
$ tslint -p ./src/app

ERROR: /Users/amatyushentsev/root/argo-cd-ui/src/app/applications/components/application-details/application-details.tsx[368, 37]: " should be '
ERROR: /Users/amatyushentsev/root/argo-cd-ui/src/app/applications/components/application-details/application-details.tsx[368, 65]: " should be '
ERROR: /Users/amatyushentsev/root/argo-cd-ui/src/app/applications/components/application-details/application-details.tsx[368, 92]: " should be '
```

you might use string interpolation:

```
const readyContainers = `${readyStatuses.length}/${allStatuses.length} containers ready`;
```

File: src/app/applications/components/application-details/application-details.tsx [hunk 4]
```diff
@@ -24,6 +24,11 @@ import * as AppUtils from '../utils';
 
 require('./application-details.scss');
 
+// ContainerStatus holds just the state of a container status, used for type-checking.
+interface ContainerStatus {
+    state: Map<string, any>;
```
Comment (alexmt) [inline src/app/applications/components/application-details/application-details.tsx]: state is not Map (Map is a class in javascript):


```
interface ContainerStatus {
  state: {[name: string]: any};
}
```

**Review by alexmt — CHANGES_REQUESTED** [review]

**Review by alexmt — APPROVED** [review]

**Pull request merged**